lib/SILGen/SILGenPattern.cpp

```diff
@@ -1031,18 +1031,27 @@ void PatternMatchEmission::emitWildcardDispatch(ClauseMatrix &clauses,
   bool hasGuard = guardExpr != nullptr;
   assert(!hasGuard || !clauses[row].isIrrefutable());
 
-  auto stmt = clauses[row].getClientData<CaseStmt>();
-  ArrayRef<CaseLabelItem> labelItems = stmt->getCaseLabelItems();
-  bool hasMultipleItems = labelItems.size() > 1;
-  
+  auto stmt = clauses[row].getClientData<Stmt>();
+  assert(isa<CaseStmt>(stmt) || isa<CatchStmt>(stmt));
+
+  bool hasMultipleItems = false;
+  if (auto *caseStmt = dyn_cast<CaseStmt>(stmt)) {
+    hasMultipleItems = caseStmt->getCaseLabelItems().size() > 1;
+  }
+
   // Bind the rest of the patterns.
   bindIrrefutablePatterns(clauses[row], args, !hasGuard, hasMultipleItems);
 
   // Emit the guard branch, if it exists.
   if (guardExpr) {
-    SGF.usingImplicitVariablesForPattern(clauses[row].getCasePattern(), stmt, [&]{
-      this->emitGuardBranch(guardExpr, guardExpr, failure);
-    });
+    if (auto *caseStmt = dyn_cast<CaseStmt>(stmt)) {
+      SGF.usingImplicitVariablesForPattern(clauses[row].getCasePattern(), caseStmt, [&]{
+        this->emitGuardBranch(guardExpr, guardExpr, failure);
+      });
+    } else {
+      // For CatchStmt
+      emitGuardBranch(guardExpr, guardExpr, failure);
```

Comment:
Thanks @jckarter .  Yes, the code is better with what you suggested and is added.